cos-qa/osf-selenium-tests
# Test project files

## Purpose

Create an automated files test we can run when an addon has undergone an aiohttp upgrade -- to catch issues early and reduce the need for manual testing.

## Summary of Changes

Added new test which tests:
* Rename (Box, Dropbox, OwnCloud, Amazon S3)
* Checkout (OSF Storage)
* Deletes (Box, Dropbox, OwnCloud, Amazon S3)
* Moves & Copies (_from_ each addon - Box, Dropbox, OwnCloud, Amazon S3 - _to_ OSF Storage)
* Download is lightly tested by a negative test (as long as an error page does not appear) (Box, Dropbox, OwnCloud, Amazon S3)

## Testing Changes Moving Forward
No major testing changes.

Note that the class FilesPage in pages/project.py is used for test_project_files.py.   The class FileWidget in components/project.py is used for tests test_file_widget_loads and test_addon_files_load in test_project.py.  In the future, we may want to put all files tests in one place.

Additionally, in the future:
- Add rename, delete tests for OSF Storage (the test currently doesn't cover this)
- Add additional move & copy tests for OSF Storage (go _from_ OSF Storage to each addon)
- Add test for folders (work has been started on it, but, was deemed out of scope for this ticket)
- Add downloads test for OSF Storage (which can positively confirm download is successful by checking the download count increased)
- Add uploads test
- Improve downloads test for the addons (probably not possible, given limitations in Selenium working with browsers)

## Ticket

https://openscience.atlassian.net/browse/ENG-685
https://openscience.atlassian.net/browse/ENG-816
https://openscience.atlassian.net/browse/ENG-817
https://openscience.atlassian.net/browse/ENG-914
https://openscience.atlassian.net/browse/ENG-915
